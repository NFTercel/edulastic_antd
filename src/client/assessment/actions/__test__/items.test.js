import {
  receiveItemsAction,
  receiveItemByIdAction,
  createItemAction,
  updateItemByIdAction,
  gotoItem,
  saveUserResponse,
  loadUserResponse
} from "../items";

describe("Items", () => {
  it("receive item should return an action", () => {
    expect(
      receiveItemsAction({
        page: 5,
        limit: 10,
        search: ""
      })
    ).toMatchSnapshot();
  });

  it("receive item by id should return an action", () => {
    expect(receiveItemByIdAction({ id: 1 })).toMatchSnapshot();
  });

  it("create item should return an action", () => {
    expect(
      createItemAction({
        rows: [
          {
            tabs: [],
            dimension: "100%",
            widgets: [
              {
                widgetType: "question",
                type: "multipleChoice",
                title: "Multiple choice",
                reference: "5c0f0c9ed8c7f4453f5bb51f",
                tabIndex: 0
              }
            ]
          }
        ],
        columns: []
      })
    ).toMatchSnapshot();
  });

  it("update item by id should return an action", () => {
    expect(updateItemByIdAction({ id: 2 })).toMatchSnapshot();
  });

  it("go to item should return an action", () => {
    expect(
      gotoItem({
        item: {
          rows: [
            {
              tabs: [],
              dimension: "100%",
              widgets: [
                {
                  widgetType: "question",
                  type: "multipleChoice",
                  title: "Multiple choice",
                  reference: "5c0f0c9ed8c7f4453f5bb51f",
                  tabIndex: 0
                }
              ]
            }
          ],
          columns: []
        }
      })
    ).toMatchSnapshot();
  });

  it("save user response should return an action", () => {
    expect(
      saveUserResponse({
        item: 2
      })
    ).toMatchSnapshot();
  });

  it("load user response should return an action", () => {
    expect(
      loadUserResponse({
        item: 3
      })
    ).toMatchSnapshot();
  });
});
