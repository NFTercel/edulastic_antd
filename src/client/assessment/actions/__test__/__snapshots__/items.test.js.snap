// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items create item should return an action 1`] = `
Object {
  "payload": Object {
    "payload": Object {
      "columns": Array [],
      "rows": Array [
        Object {
          "dimension": "100%",
          "tabs": Array [],
          "widgets": Array [
            Object {
              "reference": "5c0f0c9ed8c7f4453f5bb51f",
              "tabIndex": 0,
              "title": "Multiple choice",
              "type": "multipleChoice",
              "widgetType": "question",
            },
          ],
        },
      ],
    },
  },
  "type": "[items] create item request",
}
`;

exports[`Items go to item should return an action 1`] = `
Object {
  "payload": Object {
    "item": Object {
      "item": Object {
        "columns": Array [],
        "rows": Array [
          Object {
            "dimension": "100%",
            "tabs": Array [],
            "widgets": Array [
              Object {
                "reference": "5c0f0c9ed8c7f4453f5bb51f",
                "tabIndex": 0,
                "title": "Multiple choice",
                "type": "multipleChoice",
                "widgetType": "question",
              },
            ],
          },
        ],
      },
    },
  },
  "type": "[items] goto a particular item",
}
`;

exports[`Items load user response should return an action 1`] = `
Object {
  "payload": Object {
    "itemId": Object {
      "item": 3,
    },
  },
  "type": "[items] load user response",
}
`;

exports[`Items receive item by id should return an action 1`] = `
Object {
  "payload": Object {
    "id": Object {
      "id": 1,
    },
  },
  "type": "[items] receive item request",
}
`;

exports[`Items receive item should return an action 1`] = `
Object {
  "payload": Object {
    "limit": 10,
    "page": 5,
    "search": "",
  },
  "type": "[items] receive items request",
}
`;

exports[`Items save user response should return an action 1`] = `
Object {
  "payload": Object {
    "itemId": Object {
      "item": 2,
    },
  },
  "type": "[items] save user response",
}
`;

exports[`Items update item by id should return an action 1`] = `
Object {
  "payload": Object {
    "payload": Object {
      "id": 2,
    },
  },
  "type": "[items] update item request",
}
`;
